src/storage/store/account/link_store.rs

```diff
@@ -151,51 +127,52 @@ impl LinkStore {
                     .data
                     .as_ref()
                     .is_some_and(|data| match data.body.as_ref() {
-                        Some(message_data::Body::LinkBody(body)) => {
-                            r#type.is_empty() || body.r#type == r#type
-                        }
+                        Some(Body::LinkBody(body)) => r#type.is_empty() || body.r#type == r#type,
                         _ => false,
                     })
             }),
         )
     }
 
     pub fn get_link_compact_state_message_by_fid(
-        store: &Store,
+        store: &Store<LinkStore>,
         fid: u32,
         page_options: &PageOptions,
     ) -> Result<MessagesPage, HubError> {
         store.get_compact_state_messages_by_fid(fid, page_options)
     }
 
     pub fn get_links_by_target(
-        store: &Store,
+        store: &Store<LinkStore>,
         target: &Target,
         r#type: String,
         page_options: &PageOptions,
     ) -> Result<MessagesPage, HubError> {
-        let prefix: Vec<u8> = LinkStore::links_by_target_key(target, 0, None)?;
+        let start_prefix: Vec<u8> = LinkStore::links_by_target_key(target, 0, None)?;
+        let stop_target = match target {
+            Target::TargetFid(fid) => Target::TargetFid(fid + 1),
```

Comment:
Use increment_vec_u8, which is more generic.

Reply:
I actually preferred this method. With `increment_vec_u8` you need to be sure that the key you're incrementing is the last part of the prefix otherwise it won't do what you want. Here, I actually had `increment_vec_u8` first but then checked the key structure and saw that the target fid is somewhere in the middle of the prefix so I changed it.

Reply:
I actually just made the change you suggested in my next feature. In that one, there's some string field we index on and it was hard to figure out how to increment that. I just changed this to be consistent too.